app/src/renderer/vuex/modules/send.js

```diff
@@ -45,21 +45,27 @@ export default ({ commit, node }) => {
   }
 
   let actions = {
+    // `lock` is a Promise which is set if we are in the process
+    // of sending a transaction, so that we can ensure only one send
+    // happens at once. otherwise, we might try to send 2 transactions
+    // using the same sequence number, which means 1 of them won't be valid.
     async sendTx(...args) {
       // wait to acquire lock
       while (lock != null) {
         // eslint-disable-line no-unmodified-loop-condition
         await lock
       }
 
-      // send and unlock when done
-      lock = doSend(...args)
-
-      // wait for doSend to finish
-      let res = await lock
-      lock = null
-
-      return res
+      try {
+        // send and unlock when done
+        lock = doSend(...args)
+        // wait for doSend to finish
+        let res = await lock
+        return res
```

Comment:
This can be `return await lock`.

Reply:
I think the solution is readable enough. I will approve. If @NodeGuy still insists we should improve the code here, I am happy to create another PR.

Reply:
OK but in the future if the intention is to clarify then `res` is insufficient.  An English word is always preferable.